src/helper/widthHelper.js

```diff
@@ -12,16 +12,55 @@
 
 'use strict';
 
-import reduce from 'lodash/reduce';
+import { createSelector } from 'reselect';
 
-export function sumPropWidths(columns) {
-  return reduce(columns, (accum, column) => accum + column.props.width, 0);
-}
+const emptyArray = [];
+const emptyObject = {};
 
-export function getTotalWidth(columns) {
-  return reduce(columns, (accum, column) => accum + column.width, 0);
-}
+const cacheOptions = {
+  memoizeOptions: {
+    maxSize: 12, // 3 cell group types * 4 template types
```

Comment:
This ensures we can call the selectors for each cell group type for each template type without having to recompute the widths.